### ISSUE TITLE
Inference issues - channel pruning - with auto option - resnet/CIFAR10

### ISSUE THREAD

**Issue by florinmatei** [Tencent/PocketFlow#256]
I tried channel pruning with the "--cp_prune_option auto"  (for reinforcement learning) and we got the usual tensor flow checkpoint files. Then I converted them to a tensor flow lite file model_transformed.tflite .

For some reason when I do inference using the tensor flow lite file model_transformed.tflite I get an output that looks like this [[nan nan nan nan nan nan nan nan nan nan]] for all CIFAR10 classes.

**This is the command I used to do the original channel pruning:**
./scripts/run_local.sh nets/resnet_at_cifar10_run.py  —learner channel --batch_size_eval 64   --cp_preserve_ratio 0.5  --cp_prune_option auto  --resnet_size 20

**This is the command I used to covert the checkpoint files to tflite:**
python tools/conversion/export_chn_pruned_tflite_model.py --model_dir models_eval

When I don't use the "--cp_prune_option auto" to do channel pruning, at inference I get a normal output e.g.:
[[3.7429019e-04 5.8735928e-08 1.1391895e-05 6.4425290e-01 4.2299795e-05
  3.5514408e-01 1.7116898e-04 2.2419310e-06 1.2600908e-06 3.1574035e-07]]

Any idea why I would get  [[nan nan nan nan nan nan nan nan nan nan]] when I do inference on the models with the _auto_ channel pruning option ? 
Thank you!

**Comment by jiaxiang-wu** [comment]
Please make sure the image pre-processing (*e.g.* normalization, scaling, etc) is consistent between training and inference with *.tflite models.
Also, you may want to take a look at this issue (#258) and re-create the pruner before fine-tuning as suggested by @Nankaiming.

**Comment by florinmatei** [comment]
Thank you for the response. 

All the pre-processing is the same, I used exactly the same pipeline for channel pruning without reinforcement learning as well as for the one with reinforcement learning. As mentioned above only the one that used the reinforcement learning (with the "--cp_prune_option auto" option) gives me ten "nan" outputs when I do inference. 

I'll take a look at (#258) to see if it addressed the issues I'm seeing.
